### ISSUE TITLE
Access contents of a folder in filesystem search

### ISSUE THREAD

**Issue by djrain** [godotengine/godot-proposals#7770]
### Describe the project you are working on

Projects

### Describe the problem or limitation you are having in your project

You search for a folder in the FileSystem, like "shaders", expecting to be able to access all your shaders. Godot shows the shaders folder in the results. But you can't open it or even see what's in it. So searching for folders is nearly useless.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Folders that match the search results should be expandable to view all of their contents, just like in the default view. 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

see above

### If this enhancement will not be used often, can it be worked around with a few lines of script?

no

### Is there a reason why this should be core and not an add-on in the asset library?

it's a core feature lacking basic functionality

**Comment by jmarceno** [comment]
I've seem discussions in the past to have a better a FS Browser, but I´m no up to date about it.

Being able to actually see the files in the folder would be a step forward, but I do not know if it would not cause any other usability problems.
Like for example, if I´m looking for a file that is contained in a big folder, and match my search. Now instead of just seeing the file "shader_cool.gdshader", I´m seeing the whole "shaders" folder, which basically renders my search irrelevant.

The issue with the current file browser, is that it does not scale, like, at all. It is good for small projects, but as soon as you reach a certain size, everything falls apart, it is one of the reasons that gave-up on doing any level design inside the Editor and decided to automatize everything with Qodot.

Some of the core issues that I´ve been facing in the last 2 years.
- A long folder structure, makes that you always have a scrollbar, so if you are testing some that needs to go back and forth between 2 or 3 folders, it becomes very inefficient. (Populating textures/shaders in a material, almost always makes me face this problem)
- Not being able to have more than 1 FS open, even if read only, is another major pain point after a certain project size

I´m always shocked by not seeing people talk more about this improvement point, as it was (and still is even 2 years later) my main pain point coming from Unreal, and one the main productivity hurdles that I have.

**Comment by djrain** [comment]
> Like for example, if I´m looking for a file that is contained in a big folder, and match my search. Now instead of just seeing the file "shader_cool.gdshader", I´m seeing the whole "shaders" folder, which basically renders my search irrelevant.

Good point, there's a potential issue with files that are both in a matching folder and match on their own.

The FileSystem filtering is just that - a filtering of the file tree, not a search, which limits what can be done there. And the fact is Godot doesn't have a proper search feature anywhere, that supports folders and advanced searches.

A proper search solution is beyond the scope of this proposal. But now I'm thinking a better "quick fix" for my issue would be to allow a double-click on a folder in the filtered results, and that would simply jump to the expanded contents of that folder in the default view.

**Comment by jmarceno** [comment]
> A proper search solution is beyond the scope of this proposal. But now I'm thinking a better "quick fix" for my issue might be to allow a double-click on a folder in the filtered results, and that would simply take you to the opened folder in the default view.

In your case, doesn't the split view relieve the issue? Or you already using it? In split mode, if you delete the filter after the clicking a folder, you will stay on that folder and files will be on the bottom panel.
